### PR TITLE
feat(api explorer): adding optional handler for auth group change

### PR DESCRIPTION
**Pull request by dpim**
## 🧰 What's being changed?
* Adding support for passing an optional function to handle group change via props (which is called in addition to existing, built-in behavior).
* This function would receive the new group member id whenever onAuthGroupChange() is invoked.

🧪 Testing
* Manually verified

### CODE REVIEW

**Review by erunion — APPROVED** [review]

File: packages/api-explorer/src/index.jsx [hunk 1]
```diff
@@ -114,6 +114,10 @@ class ApiExplorer extends React.Component {
    * @param {string} group
    */
   onAuthGroupChange(group) {
+    // Invoke supplementary handler (if available)
+    if (this.props.onAuthGroupChange && typeof this.props.onAuthGroupChange === 'function') {
```
[the remark this answers is not in the document]
Comment (Dashron) [reply packages/api-explorer/src/index.jsx:118]: I like that a lot, always a fan of abusing empty functions. 

Also, as far as I can tell from reading up on PropTypes, I'm not sure if the typeof flag was ever necessary thanks to the function type definition below.

**Review by Dashron — CHANGES_REQUESTED** [review]
lmk what you think of Jon and my comments.

File: packages/api-explorer/src/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dpim) [reply packages/api-explorer/src/index.jsx]: SGTM - refactored to use default props

**Review by erunion — APPROVED** [review]

**Comment by erunion** [comment]
@dpim I'll cut a release with this later this week for you.

**Review by Dashron — COMMENTED** [review]
👍

**Pull request merged**